lib/collection/src/operations/point_ops.rs

```diff
@@ -65,14 +55,77 @@ pub enum PointsSelector {
     FilterSelector(FilterSelector),
 }
 
-#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone)]
+// Structure used for deriving custom JsonSchema only
+#[derive(Debug, Deserialize, Serialize, JsonSchema)]
+struct PointsList {
+    points: Vec<PointStruct>,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
 #[serde(rename_all = "snake_case")]
-#[serde(untagged)]
 pub enum PointInsertOperations {
     /// Inset points from a batch.
-    PointsBatch(PointsBatch),
+    #[serde(rename = "batch")]
+    PointsBatch(Batch),
     /// Insert points from a list
-    PointsList(PointsList),
+    #[serde(rename = "points")]
+    PointsList(Vec<PointStruct>),
+}
+
+impl JsonSchema for PointInsertOperations {
```

Comment:
how did you validate the correctness of the generated schema?

Reply:
With integration tests here https://github.com/qdrant/qdrant/tree/master/openapi/tests/openapi_integration